src/main/java/xyz/nucleoid/bedwars/game/active/shop/BwItemShop.java

```diff
@@ -12,104 +12,254 @@
 import xyz.nucleoid.plasmid.shop.ShopUi;
 import xyz.nucleoid.plasmid.util.ColoredBlocks;
 import xyz.nucleoid.plasmid.util.ItemStackBuilder;
+
 import net.minecraft.block.Blocks;
+import net.minecraft.enchantment.Enchantment;
 import net.minecraft.enchantment.Enchantments;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.potion.PotionUtil;
+import net.minecraft.potion.Potions;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.text.LiteralText;
 import net.minecraft.text.Text;
 import net.minecraft.util.DyeColor;
 
 public final class BwItemShop {
-    //TODO: reorganize shop layout
     public static ShopUi create(ServerPlayerEntity player, BwActive game) {
-        return ShopUi.create(new LiteralText("Item Shop"), shop -> {
+        //TODO User preferred items page
+        return createBlocks(player, game);
+        //FIXME Item sorting, through Plasmid Shop
+    }
+
+    private static void addNavbar(ShopBuilder shop, ServerPlayerEntity player, BwActive game, int pageIndex) {
+        // Creates a page navigation bar at the top of the shop
+        if(pageIndex==1){
```

Comment:
There's a lot of duplication here that could be cleaned up with a helper function like:
`addNavigationTab(shop, Blocks.END_STONE, new LiteralText("Blocks"), BwItemShop::createBlocks, pageIndex == 1)`

Reply:
nitpick on the same line for formatting (`if (pageIndex == 1) {` is better)

Reply:
@Gegy Any ideas on how to implement that?